### PR TITLE
Website updates (documentation build)

### PR DESCRIPTION
**Pull request by RoseZhao**
I worked with @stefanv to improve various aspects of the scikit-image website.  The changes made are listed in more detail below.  Overall, improvements are aimed at visual consistency, making version numbers prominent where relevant, and better navigation.

This pull request updates the docs build.  Also see https://github.com/scikit-image/scikit-image-web/pull/46 for accompanying changes to scikit-image-web.

1. Delete "Navigation", "Previous" and "Next" from documentation. 
2. Fix the alignment issue with Python and Jupyter notebook download in gallery examples (the buttons appeared underneath, instead of next to one another) and change their color to blue (from weird yellow).
3. Change the title of "API Reference" to "API Reference for scikit-image Version x.y".
4. For each entry listed in the gallery page, remove the gradient (more modern look) and add a dark gray border around it.
5. For each topic in the user guide, add a table of contents in the sidebar.
6. Remove gradient of gallery entries on mouse hover.
7. For each topic in API Reference, move the "source" button to the very right and enlarge the sentence next to it to make the version very clear.
8. Add a hyperlink to "Download for Development" that leads to "Installation from Source".
9. List all submodules in skimage at the top of API Reference.
10. Remove "Versions" sidebar and add a prominent block with current version and a link to scikit-image/docs for other versions.

### CODE REVIEW

**Comment by soupault** [comment]
@RoseZhao Could you provide a description for this PR?

**Comment by stefanv** [comment]
@RoseZhao Can you please update this pull request with all the changes made, and list them in the PR description?  Thanks!

File: doc/release/_release_notes_for_docs.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
+We're happy to announce the release of scikit-image v0.12!
```
Comment (stefanv) [inline doc/release/_release_notes_for_docs.rst]: We should rebase on master and get rid of this file.

File: doc/tools/apigen.py [hunk 2]
```diff
@@ -457,9 +457,24 @@ def write_index(self, outdir, froot='gen', relative_to=None):
         w = idx.write
         w('.. AUTO-GENERATED FILE -- DO NOT EDIT!\n\n')
 
-        title = "API Reference"
+        title = "API Reference for scikit-image version 0.12.3"
```
Comment (stefanv) [inline doc/tools/apigen.py]: This version needs to change dynamically depending on the version of the docs built.

File: doc/tools/apigen.py [hunk 3]
```diff
@@ -457,9 +457,24 @@ def write_index(self, outdir, froot='gen', relative_to=None):
...
         w(title + "\n")
         w("=" * len(title) + "\n\n")
+
+        for f in self.written_modules:
+            module_name = f.split('.')
```
Comment (stefanv) [inline doc/tools/apigen.py]: Let's add a comment here to show what's happening:

```
# We look at the module name.  If it is `skimage`, display, if `skimage.submodule`, only show `submodule`,
# if it is `skimage.submodule.subsubmodule`, ignore.
```

Also, let's change the logic below slightly, to something like this:

```
if len(module_name) > 2:
    continue
elif len(module_name) == 1:
    display_name = module_name[0]
elif len(module_name) == 2:
    display_name = module_name[1]

w('- `{0} <{1}.html`__\n'.format(os.path.join(display_name, module_name)))
```

File: doc/release/_release_notes_for_docs.rst (hunk 1 above)
Comment (RoseZhao) [inline doc/release/_release_notes_for_docs.rst]: Done.

**Comment by stefanv** [comment]
I'm going to merge these stylistic changes to the docs & website; please take a look at the built version online and let me know if we should change anything further.

**Pull request merged**